### PR TITLE
Generate bin scripts via setuptools entry_points

### PR DESCRIPTION
**Pull request by kahowell**
This does a lot for us, but especially lets the python used to run setup
automatically be the one used to run the scripts (important for Python 3
compatibility).

### CODE REVIEW

**Comment by kahowell** [comment]
I developed this on top of #1740 ... Some of the fixes in there will satisfy failures here, especially Python 3 syntax issues.

File: Makefile [hunk 1]
```diff
@@ -120,10 +120,14 @@ clean:
 	$(PYTHON) ./setup.py clean --all
 	rm -rf cover/ htmlcov/ docs/sphinx/_build/ build/ dist/
 
-rhsmcertd: $(DAEMONS_SRC_DIR)/rhsmcertd.c
+.PHONY: mkdir-bin
+mkdir-bin:
+	mkdir bin
```
Comment (jirihnidek) [inline Makefile]: When `./bin` directory already exists, then the command fails.

File: Makefile [hunk 2]
```diff
@@ -277,6 +280,15 @@ install-post-boot: install-firstboot install-initial-setup
...
 	$(PYTHON) ./setup.py install --root $(PREFIX) --gtk-version=$(GTK_VERSION) --rpm-version=$(VERSION) --with-systemd=$(WITH_SYSTEMD) --prefix=/usr
+	mkdir -p $(PREFIX)/usr/sbin/
+	mkdir -p $(PREFIX)/$(LIBEXEC_DIR)/
+	mv $(PREFIX)/usr/bin/subscription-manager $(PREFIX)/usr/sbin/
+	mv $(PREFIX)/usr/bin/subscription-manager-gui $(PREFIX)/usr/sbin/
+	mv $(PREFIX)/usr/bin/rhn-migrate-classic-to-rhsm $(PREFIX)/usr/sbin/
+	mv $(PREFIX)/usr/bin/rhsmcertd-worker $(PREFIX)/$(LIBEXEC_DIR)/
+	mv $(PREFIX)/usr/bin/rhsm-service $(PREFIX)/$(LIBEXEC_DIR)/
+	mv $(PREFIX)/usr/bin/rhsm-facts-service $(PREFIX)/$(LIBEXEC_DIR)/
+	mv $(PREFIX)/usr/bin/rhsmd $(PREFIX)/$(LIBEXEC_DIR)/
```
Comment (jirihnidek) [inline Makefile]: BTW: What is intention of the target `install-via-setup`? When I tried to install it using `make install-via-setup`, then I get following error:

```
[root@fedora ~]# subscription-manager version
Traceback (most recent call last):
  File "/usr/sbin/subscription-manager", line 11, in <module>
    load_entry_point('subscription-manager==1.20.7', 'gui_scripts', 'subscription-manager-gui')()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 563, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2651, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2305, in load
    return self.resolve()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2311, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib64/python2.7/site-packages/subscription_manager/scripts/subscription_manager_gui.py", line 124, in <module>
    from subscription_manager.ga import gtk_compat
  File "/usr/lib64/python2.7/site-packages/subscription_manager/ga_loader.py", line 118, in load_module
    return self._import_real_module(fullname, real_module_name, real_module_from)
  File "/usr/lib64/python2.7/site-packages/subscription_manager/ga_loader.py", line 121, in _import_real_module
    ret = __import__(module_name, globals(), locals(), [module_from])
ImportError: No module named ga_impls
```

File: Makefile (hunk 1 above)
Comment (mdellweg) [reply Makefile]: `mkdir -p` should help here.

Comment (kahowell) [reply Makefile]: @mdellweg thanks for the suggestion! Will do.

File: Makefile (hunk 2 above)
Comment (kahowell) [reply Makefile]: Unfortunately, our overall build/install process is a bit scattered... Sequentially we do this in the specfile:

1. In `setup.py`, we exclude the ga_impls (see https://github.com/candlepin/subscription-manager/blob/10e86e0de3fd1c705f7a87a735611c2e4282232d/setup.py#L311)
2. After setup.py install is ran, we have no ga_impls landed on the target system. So we then add the appropriate one (see https://github.com/candlepin/subscription-manager/blob/10e86e0de3fd1c705f7a87a735611c2e4282232d/subscription-manager.spec#L577-L582) (GTK2 or GTK3 depending on the build target).

So due to this, `make install` is not sufficient to actually completely install subscription-manager. I consider this technical debt... Ideally the conditional installation would be in the Makefile and *not* the specfile, and we'd have a much simpler specfile. I think it's bad to have so much of our build/install coupled to RPM packaging.

Comment (mdellweg) [reply Makefile]: That would be nice. Especially for packaging in other ecosystems. ;)

Comment (kahowell) [reply Makefile]: Note: I edited the previous comment to prevent confusion. Please reread if necessary. Apologies for any confusion.

**Comment by kahowell** [comment]
@jirihnidek I left some notes in the README about local development (including why `python src/subscription_manager/scripts/subscription_manager.py` doesn't work).

File: Makefile (hunk 2 above)
Comment (kahowell) [reply Makefile]: @mdellweg, when I first looked at the spec file, I thought that it was landing the appropriate ga_impl. I misread though. The `Makefile` *does* land the appropriate ga_impl (see updated comment).

Comment (mdellweg) [reply Makefile]: So the `make install` is sufficient. Thanks for clearing up.

**Comment by kahowell** [comment]
f8aa2e43..cb374ec9: rebase against master & squash (no conflicts 😄)

**Pull request merged**